File: src/main/java/org/chousik/collection/MusicGenre.java
package org.chousik.collection;

import java.util.Arrays;
import java.util.LinkedList;
import java.util.stream.Collectors;

/**
 * Класс перечисление жанров музыки
 */
public enum MusicGenre {
    JAZZ,
    SOUL,
    POP;

    /**
     * @return возвращает все значения перечисления
     */
    public static String getValue() {
        return Arrays.stream(MusicGenre.values())
                .map(MusicGenre::name)
                .collect(Collectors.joining(", "));
    }
}

File: src/main/kotlin/collection/MusicGenre.kt
package org.chousik.collection

import java.util.*
import java.util.stream.Collectors


enum class MusicGenre {
    JAZZ,
    SOUL,
    POP;

    companion object {
        val value: String

            get() = Arrays.stream(entries.toTypedArray())
                .map({ obj: MusicGenre -> obj.name })
                .collect(Collectors.joining(", "))
    }
}